app/views/view_components/_preview_source.html.erb

```diff
@@ -7,9 +7,13 @@
         <%= h @preview.preview_source(@example_name) %>
       </code>
     <% else %>
-      <% template = @view_renderer.lookup_context.find_template(@render_args[:template]) %>
+      <% template = find_template_source(lookup_context: @view_renderer.lookup_context, template_identifier: @render_args[:template]) %>
       <code class="language-<%= prism_language_name(template: template) %>">
-        <%= h template.source %>
+        <% if template.respond_to?(:source) %>
+          <%= h template.source %>
+        <% else %>
+          <%= h File.read(template) %>
```

Comment:
I'am looking for feedback if this is an ok approach. I've never really used `File` in a template before and don't know if we see any consequences. Based on how this is written, I think it isn't an issue but just wanted to make sure.

Reply:
This makes me think that we should change `find_template_source` to simply return the source of the template, and not an object or file path as it appears to now. What do you think?

Reply:
@joelhawksley are you suggesting that we wouldn't even need to do `template.source` like in line 13? I think that might add an extra layer of complexity since the method to determine the extension or type is expecting an object.

I'am going to try to modify it in such a way that might be more verbose but clearer in the end. I do think we shouldn't cause any confusion about wither or not `template` is an object or a file path. 

I'll see what I can do in terms of simplifying

Reply:
@joelhawksley hey there! I've made some changes to simplify things. My main goal was to reduce the complexity of the template and try to arrange it in the helper method. This way -- looking at the preview template doesn't require the reader to know exactly how the template source or language is being derived.

Let me know how this looks